arroyo-console/src/routes/connections/JsonForm.tsx

```diff
@@ -150,6 +156,151 @@ function NumberWidget({
   );
 }
 
+function AutocompleteWidget({
+  path,
+  title,
+  description,
+  placeholder,
+  required,
+  value,
+  errors,
+  onChange,
+  readonly,
+  autocompleteData,
+  autocompleteError,
+}: {
+  path: string;
+  title: string;
+  description?: string;
+  placeholder?: string;
+  required?: boolean;
+  value: string;
+  errors: any;
+  onChange: (e: React.ChangeEvent<any>) => void;
+  readonly?: boolean;
+  autocompleteData?: {
+    values: {
+      [key: string]: string[] | undefined;
+    };
+  };
+  autocompleteError?: string;
+}) {
+  const ourItems = autocompleteData?.values[path] || [];
+  const [allItems, setAllItems] = React.useState<string[]>(ourItems);
+  const [items, setItems] = React.useState<string[]>(ourItems);
+
+  useEffect(() => {
+    if (autocompleteData && allItems.length == 0) {
+      setAllItems(ourItems);
+      setItems(ourItems);
+    }
+  }, [autocompleteData]);
+
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    highlightedIndex,
+    getItemProps,
+  } = useCombobox({
+    onInputValueChange({ inputValue }) {
+      // @ts-ignore
+      onChange({ target: { name: path, value: inputValue } });
+      if (inputValue && inputValue?.length > 0) {
+        setItems(
+          allItems.filter(items => items.toLowerCase().startsWith(inputValue.toLowerCase()))
```

Comment:
```suggestion
          allItems.filter(items => items.toLowerCase().includes(inputValue.toLowerCase()))
```

for a more lenient search